src/backend/utils/text-parser.js

```diff
@@ -1,20 +1,28 @@
-const puppeteer = require('puppeteer');
+const { Cluster } = require('puppeteer-cluster');
 
-module.exports = async function(htmlFragment) {
-  let browser;
-  try {
-    browser = await puppeteer.launch();
-    const page = await browser.newPage();
+let isInitialized = false;
+let cluster;
 
-    await page.setContent(htmlFragment, {
-      waitUntil: 'domcontentloaded',
+module.exports.parse = async function(htmlFragment) {
+  if (!isInitialized) {
```

Comment:
Flip this logic around, so you can return early:

```js
if(isInitialized) {
  return cluster.execute(htmlFragment);
}
// rest of code here...
```

Reply:
Also, do we not need to `await cluster.execute(...)`?  It returns a Promise.

Reply:
will be fixed shortly. same fix will also be applied to `await page.evaluate`